### ISSUE TITLE
Notification Bar Icon Doesn't appear if I pass setenv argument

### ISSUE THREAD

**Issue by melink14** [rupor-github/wsl-ssh-agent#1]
If I open it with no arguments the 'keys' icon appears normally but env isn't updated.

If I run from cmd prompt, with -setenv the env is configured but no icon appears.

**Comment by rupor-github** [comment]
You may want to look into Windows->Settings->Taskbar->"Select which icons apper of taskbar" and make sure that wsl-ssh-agent-gui.exe is "on". Windows would move your icon into hide area under "V" icon on the taskbar rather quickly. I cannot explain why you would not see the icon otherwise.

**Comment by melink14** [comment]
Thanks. I definitely understand that sometimes I need to check in the expanded view for notification icons. (I use many utilities and tools that get put there.)

It's pretty easy to repro, but not sure if there's any other information I can give to help debug.

**Comment by rupor-github** [comment]
It has been very long time since I even tried to understand how Windows does what it does. As far as I could remember notification area icon gets cached in temporary directory and I have no real idea why it would not be shown. I have seen something similar once on my laptop after hibernation - rebooted immediately and it started to work normally. Try setting "TRACE=true" in your environment and collect detailed log using https://docs.microsoft.com/en-us/sysinternals/downloads/debugview , but I really doubt it will help with visuals. Using --debug flag (if it would not interfere with repro would not hurt either)

**Comment by rupor-github** [comment]
Here is another idea (may be a bit crazy, I have not tried this): https://www.softperfect.com/contact/knowledgebase.php?article=11

**Comment by melink14** [comment]
I'll try your debug tips, thanks.

Also, I did more debugging and it seems like the icon was not the only problem.

Even though the exe is running (as seen in task manager) it doesn't work (can't ssh to servers using it) even though the environment variables seem to work.

If I run without --setenv and manually update the environment variables manually then everything works and the icon is there. I wonder if it's hitting an error and hanging. (the socket and lock files are created in my Temp directory at least)

**Comment by melink14** [comment]
Ran DebugView and ran:
`C:\wsl-ssh-agent>wsl-ssh-agent-gui.exe --setenv --debug`

[17400] [wsl-ssh-agent-gui] Windows - CurrentMajorVersionNumber: 10, CurrentBuild: 18362
[17400] [wsl-ssh-agent-gui] Set 'SSH_AUTH_SOCK=C:\Users\<omit>\AppData\Local\Temp\ssh-321667491.sock'
[17400] [wsl-ssh-agent-gui] Was 'WSLENV=:SSH_AUTH_SOCK/up'
[17400] [wsl-ssh-agent-gui] Set 'WSLENV=:SSH_AUTH_SOCK/up'

Compared to without setenv:
`C:\wsl-ssh-agent>wsl-ssh-agent-gui.exe --debug`
[4156] [wsl-ssh-agent-gui] Windows - CurrentMajorVersionNumber: 10, CurrentBuild: 18362
[4156] [wsl-ssh-agent-gui] Listening on Unix socket: C:\Users\<omit>\AppData\Local\Temp\ssh-260504107.sock


Clearly, it's never getting ot the 'Listening...' part when running with setenv though not sure why.

**Comment by rupor-github** [comment]
In my case I do not have "Was 'WSLENV=:SSH_AUTH_SOCK/up'" - it is being added by the program. So may be (just may be) in this case SSH_AUTH_SOCK is not propagated. I will try to reproduce this locally - may learn something. I do not understand how it may prevent program from running...

**Comment by rupor-github** [comment]
Also - I do not understand why do you have ":" at the beginning of WSLENV? Something fishy is going on with this setting - in my case:
```
[12660] [wsl-ssh-agent-gui] Was 'WSLENV=USERPROFILE/up'
[12660] [wsl-ssh-agent-gui] Set 'WSLENV=USERPROFILE/up:SSH_AUTH_SOCK/up'
[12660] [wsl-ssh-agent-gui] Listening on Unix socket: C:\Users\rupor\AppData\Local\Temp\ssh-437530271.sock
```

":" should separate variables, how did it get at the beginning? But even if I try to do that, I still have it working:

```
[9364] [wsl-ssh-agent-gui] Was 'WSLENV=:SSH_AUTH_SOCK/up'
[9364] [wsl-ssh-agent-gui] Set 'WSLENV=:SSH_AUTH_SOCK/up'
[9364] [wsl-ssh-agent-gui] Listening on Unix socket: C:\Users\rupor\AppData\Local\Temp\ssh-236165383.sock
```
Are you sure that there no previous instances of application hunging somewhere in the backgroung showing error MessageBox?

**Comment by melink14** [comment]
My guess is perhaps the leading `:` is caused by having no WSLENV variable set before running the program? (or perhaps an empty one? may some edge case in the split and join logic there.)

I see that any error is supposed to pop up a message box but I definitely don't see any. (and the program doesn't terminate...)

Maybe one of the syscalls isn't returning...
I may add some more logging on my own if I'm curious (since only I can repro) but until then, I'll probably just use the `--socket` flag with a file in a known location.

**Comment by rupor-github** [comment]
I'll recheck logic again - please, keep me posted

**Comment by rupor-github** [comment]
@melink14 - latest version cleaned up logic in setenv, no more ":" and no empty WSLENV left behind. You may want to manually remove old "improper" env variables if any.

**Comment by scragz** [comment]
Same here with most recent version, if I pass setenv it won't show in tray.

**Comment by rupor-github** [comment]
Would you mind sharing details about Windows version you are running? Any antivirus software?

**Comment by scragz** [comment]
Windows 10 Version 1803 Build 17134.1006. No antivirus.

**Comment by rupor-github** [comment]
Please, read first page carefully:

`wsl-ssh-agent-gui was tested on Windows 10 1903 with multiple distributions and should work on anything starting with 1809 - beginning with insider build 17063 and would not work on older versions of Windows 10, because it requires AF_UNIX socket support feature.
`

I do check for build number when program starts and it should show message box. It is entirely possible however that I am missing something in this check - it was following golang run-time check for sockets which may be outdated: `return m >= 10 && b >= 17063`.  I will add checking for release later... Answer in your case - 1803 is not a supported Windows version.

**Comment by scragz** [comment]
The socket stuff works! Just saying that like OP that passing setenv the tray icon doesn't show, unrelated to any of the core socket functionality. I understand I'm not technically on a supported version, just trying to help you figure the bug out.

**Comment by rupor-github** [comment]
I do understand - wish I could repo it somehow, it is a bit non-transparent. All I could suggest is trying to clean up icon cache in Windows (https://www.softperfect.com/contact/knowledgebase.php?article=11) it seems harmless enough and see if it helps :) Have no real idea what would cause this behavior, but suspect some call from go code to Windows API is failing silently somewhere...

**Comment by rupor-github** [comment]
This may be connected to Issue #8 and may be fixed in the version 1.4
